server/src/server.ts

```diff
@@ -74,6 +78,127 @@ connection.onInitialize(
     },
 );
 
+connection.onInitialized((_params) => {
+    connection.workspace.getWorkspaceFolders().then((_value) => {
+        const workspacePath = url.fileURLToPath(_value[0].uri);
+        glob('**/project.kao', {
+            cwd: workspacePath,
+        })
+            .then((_matches) => {
+                for (const filePath of _matches) {
+                    if (openProjectFile(workspacePath, filePath)) {
+                        // There should be only one `project.kao` file. If we found a good candidate, stop the loop.
+                        return;
+                    }
+                }
+            })
+            .catch((_error) => console.error(_error));
+        /*
+         * List of the file-extensions of yseopml files that are never set by the user in `project.kao`-like files.
+         * This list is a subset of the file-extensions known by this extension as set in `client/package.json`.
+         */
+        const yseopmlExtensions = ['yclass', 'yobject', 'ycomplete'];
+        for (const extension of yseopmlExtensions) {
+            parseFilesWithExtension(workspacePath, extension);
+        }
+    });
+});
+
+/**
+ * Find all the files in the workspace that have the extension `extension`
+ * and open them as `TextDocument` objects. This will result in a parsing request for
+ * these files and have it known by the extension.
+ * This function excludes the results from `.generated-yml/`.
+ *
+ * @param extension The extension of the files to look for
+ */
+function parseFilesWithExtension(workspacePath: string, extension: string): void {
+    glob(`**/*.${extension}`, {
+        cwd: workspacePath,
+        ignore: '**/.generated-yml/**',
+    })
+        .then((_matches) => {
+            if (!_matches) {
+                return;
```

Comment:
1) I’m surprised to see there are nice `cwd` and `ignore` parameters to these nice globbing tools but no way to immediately discard items that do not match.
2) Btw, are we sure we’re not going to try to open directories as if they were regular files? Did you try creating weird dirs like `foo.<extension>/`?
3) Btw-2, could it be optimized by calling it only once, with multiple extensions? I dunno if these tools allow patterns like `foo.{yml,yclass,yobject}`. That would for example factorize the `ignore: …` checks. But maybe this would not be significant. I dunno. I tried to see how you were using that function (e.g., if you were calling it multiple times in consecutive lines or if these calls were spread out more broadly) but it seems to be either in collapsed areas of the diff or even outside of it. :( If you like this idea and think it could be applicable, you can always create a GitHub ticket for it or something, I guess.

Reply:
1. It is completely related to one of your other comments. I wrongly supposed that `[]` was falsy. I'll remove this.
2. Not sure. I'll add a check in the forEach.
3. If `foo.{yml,yclass,yobject}` is a correct glob expression, it should be OK and very helpful indeed. I use this function only during the server initialization and loop over the desired extensions.